src/google/protobuf/util/field_mask_util.cc

```diff
@@ -417,14 +416,14 @@ void FieldMaskTree::IntersectPath(const std::string& path, FieldMaskTree* out) {
     return;
   }
   const Node* node = &root_;
-  for (int i = 0; i < parts.size(); ++i) {
+  for (auto && part : parts) {
     if (node->children.empty()) {
       if (node != &root_) {
         out->AddPath(path);
       }
       return;
     }
-    const std::string& node_name = parts[i];
+    const std::string& node_name = part;
```

Comment:
I would just call the loop variable `node_name` so that this line becomes unnecessary.

Reply:
Done as well.